src/components/ItemFilters/ItemFilters.tsx

```diff
@@ -137,28 +132,27 @@ const ItemFilters = ({
 
   // function passed to filterElements map callback for generating renderMultiSelect prop of MultiSelectGroup
   const renderMultiSelect = ({ isBlockElement, multiSelectWidth }) =>
-    filterCheckboxGroups.map((checkboxGroup: FilterCheckboxGroup) =>
-      checkboxGroup?.items.length ? (
+    filterData.map((itemFilterData: ItemFilterData) => {
+      const checkboxGroup = itemFilterData.formattedFilterData
+      return checkboxGroup?.items.length ? (
         <MultiSelect
           buttonText={checkboxGroup.name}
-          id={`${checkboxGroup.id}-multi-select`}
+          id={`${checkboxGroup.id}`}
           data-testid={`${checkboxGroup.id}-multi-select`}
           items={checkboxGroup.items}
           key={checkboxGroup.id}
-          onChange={(e: React.ChangeEvent<HTMLInputElement>): void => {
-            console.log(e.target.value)
+          onChange={async (e: React.ChangeEvent<HTMLInputElement>) => {
+            await handleMultiSelectChange(e.target.id, checkboxGroup.id)
           }}
-          onClear={() => {
```

Comment:
You do need the `onClear` function. Right now the count button with the "X" does not work. You can click on it but it doesn't remove the selected filters.

Reply:
You probably need to use  `handleRemoveFilter` here.

Reply:
great catch! had to add a new callback (handleClearFilterGroup) since handleRemoveFilter finds and removes a single filter value.